sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,265 @@
+"""Stacked ensembles"""
+
+# Author: Caio Oliveira <caioaao@gmail.com>
+# License: BSD 3 clause
+
+from ..base import (BaseEstimator, TransformerMixin, MetaEstimatorMixin, clone)
+from ..model_selection import cross_val_predict
+from ..pipeline import FeatureUnion
+from ..preprocessing import FunctionTransformer
+
+
+class StackingTransformer(BaseEstimator, MetaEstimatorMixin, TransformerMixin):
+    """Transformer to turn estimators into meta-estimators for model stacking
+
+    In stacked generalization, meta estimators are combined in layers to
+    improve the final result. To prevent data leaks between layers, a procedure
+    similar to cross validation is adopted, where the model is trained in one
+    part of the set and predicts the other part. In ``StackingTransformer``, it
+    happens during ``fit_transform``, as the result of this procedure is what
+    should be used by the next layers. Note that this behavior is different
+    from ``fit().transform()``. Read more in the
+    :ref:`User Guide <stacking_transformer>`.
+
+    Parameters
+    ----------
+    estimator : predictor
+        The estimator to be blended.
+
+    cv : int, cross-validation generator or an iterable, optional (default=3)
+        Determines the cross-validation splitting strategy to be used for
+        generating features to train the next layer on the stacked ensemble or,
+        more specifically, during ``fit_transform``.
+
+        Possible inputs for cv are:
+
+        - None, to use the default 3-fold cross-validation,
+        - integer, to specify the number of folds.
+        - An object to be used as a cross-validation generator.
+        - An iterable yielding train/test splits.
+
+        For integer/None inputs, if the estimator is a classifier and ``y`` is
+        either binary or multiclass,
+        :class:`sklearn.model_selection.StratifiedKFold` is used. In all
+        other cases, :class:`sklearn.model_selection.KFold` is used.
+
+        Refer :ref:`User Guide <cross_validation>` for the various
+        cross-validation strategies that can be used here.
+
+    method : string, optional (default='auto')
```

Comment:
@jnothman Is is a big deal that this is different from `cross_val_predict` method argument. Do we want to enforce the same behaviour.